picard/ui/metadatabox.py

```diff
@@ -467,6 +469,21 @@ def _update_tags(self):
         if not (files or tracks):
             return None
 
+        while not new_selection:  # Just an if with multiple exit points
+            # If we are dealing with the same selection
+            # skip updates unless it we are dealing with a single file/track
+            single_file = len(files) == 1
+            single_track = len(tracks) == 1
+            if single_file or single_track:
+                break
+
+            # Or if we are dealing with a single cluster/album
+            single_file_album = len(set([file.metadata["album"] for file in files])) == 1
+            single_track_album = len(set([track.metadata["album"] for track in tracks])) == 1
+            if single_file_album or single_track_album:
+                break
```

Comment:
I think we can avoid some calculations rewriting it:

```python
        while not new_selection:  # Just an if with multiple exit points
            # If we are dealing with the same selection
            # skip updates unless it we are dealing with a single file/track
            if len(files) == 1:
                break
            if len(tracks) == 1:
                break
            # Or if we are dealing with a single cluster/album
            single_file_album = len(set([file.metadata["album"] for file in files])) == 1
            if single_file_album:
                break
            single_track_album = len(set([track.metadata["album"] for track in tracks])) == 1
            if single_track_album:
                break
            return self.tag_diff
```

I didn't profile though.

Reply:
We sure can. I'm going to change it.

Reply:
I've made a few changes to cache these results and it is a bit faster. It looks kinda bad to me style-wise, but not sure how you, @phw or @Sophist-UK would do it.